draft-ietf-taps-interface.md

```diff
@@ -1669,12 +1685,55 @@ Type:
 Default:
 : Disabled
 
-This property specifies how long to wait before deciding that a Connection has
-failed when trying to reliably deliver data to the destination. Adjusting this Property
+This property specifies how long to wait before deciding that an active Connection has
+failed when trying to reliably deliver data to the Remote Endpoint. Adjusting this Property
 will only take effect when the underlying stack supports reliability. The special value
 `Disabled` means that this timeout is not scheduled to happen. This can be a valid
 choice with unreliable data transfer (e.g., when UDP is the underlying transport protocol).
+See also {{keep-alive-timeout}}.
+
+### Timeout for keep alive packets {#keep-alive-timeout}
+
+Name:
+: keepaliveTimeout
+
+Type:
+: Numeric, with special value `Disabled`
+
+Default:
+: Disabled
+
+A transport system can be requested a protcol that supports sending keep alive packets {{keep-alive}}.
+This property specifies the maximum time an idle connection (one for which no transport
+packets have been sent) should wait before 
+the Local Endpoint sends a keep-alive packet to the Remote Endpoint. Adjusting this Property
+will only take effect when the underlying stack supports sending keep-alive packet. The special value
+`Disabled` means that this timeout is not scheduled to happen, and does not request the
+transport system to send keep-alive packets. Guidance on setting this value is 
+provided in {{!BCP145}}.
+
+### Limit for Keepalives {#keep-alive-count}
 
+Name:
+: keepaliveCount
+
+Type:
+: Numeric, with special value `Disabled`
+
+Default:
+: 5
+
+Some protocols track responses to the sent keep-alive packets that are returned by the remote
```

Comment:
This sentence makes me convince even more that we should leave this one to protocol specific parameters.

Reply:
That might be OK, but then it means the App needs to say if you use SCTP, do this, if you use TCP do that, if you use STUN do this, etc. I suspect this becomes messy very quickly.

Reply:
Yes, I think the intention is to design a generic, protocol-independent type of keep-alive here?  I certainly don't think "if you use SCTP do this, if you use TCP do that" etc. is how this should become.